<launch>
  <arg name="input/rois_number" default="1"/>
  <arg name="input/rois0" default="rois0"/>
  <arg name="input/camera_info0" default="/camera_info0"/>
  <arg name="input/rois1" default="rois1"/>
  <arg name="input/camera_info1" default="/camera_info1"/>
  <arg name="input/rois2" default="rois2"/>
  <arg name="input/camera_info2" default="/camera_info2"/>
  <arg name="input/rois3" default="rois3"/>
  <arg name="input/camera_info3" default="/camera_info3"/>
  <arg name="input/rois4" default="rois4"/>
  <arg name="input/camera_info4" default="/camera_info4"/>
  <arg name="input/rois5" default="rois5"/>
  <arg name="input/camera_info5" default="/camera_info5"/>
  <arg name="input/rois6" default="rois6"/>
  <arg name="input/camera_info6" default="/camera_info6"/>
  <arg name="input/rois7" default="rois7"/>
  <arg name="input/camera_info7" default="/camera_info7"/>
  <arg name="input/clusters" default="clusters"/>
  <arg name="output/clusters" default="labeled_clusters"/>
  <arg name="sync_param_path" default="$(find-pkg-share image_projection_based_fusion)/config/roi_sync.param.yaml"/>
  <arg name="remove_unknown" default="true"/>
  <arg name="trust_distance" default="100.0"/>

  <!-- for eval variable-->
  <arg name="input_rois_number" default="$(var input/rois_number)"/>

  <!-- debug -->
  <!-- cspell: ignore minx, maxx, miny, maxy, minz, maxz  -->
  <arg name="debug_mode" default="false"/>
  <arg name="filter_scope_minx" default="-100"/>
  <arg name="filter_scope_maxx" default="100"/>
  <arg name="filter_scope_miny" default="-100"/>
  <arg name="filter_scope_maxy" default="100"/>
  <arg name="filter_scope_minz" default="-100"/>
  <arg name="filter_scope_maxz" default="100"/>
  <arg name="image_buffer_size" default="15"/>
  <arg name="input/image0" default="/image_raw0"/>
  <arg name="input/image1" default="/image_raw1"/>
  <arg name="input/image2" default="/image_raw2"/>
  <arg name="input/image3" default="/image_raw3"/>
  <arg name="input/image4" default="/image_raw4"/>
  <arg name="input/image5" default="/image_raw5"/>
  <arg name="input/image6" default="/image_raw6"/>
  <arg name="input/image7" default="/image_raw7"/>
  <group>
    <node pkg="image_projection_based_fusion" exec="roi_cluster_fusion_node" name="roi_cluster_fusion" output="screen">
      <param name="use_iou" value="true"/>
      <param name="use_iou_x" value="false"/>
      <param name="use_iou_y" value="false"/>
      <param name="only_allow_inside_cluster" value="true"/>
      <param name="roi_scale_factor" value="1.1"/>
      <param name="iou_threshold" value="0.35"/>
      <param name="rois_number" value="$(var input/rois_number)"/>
      <param name="remove_unknown" value="$(var remove_unknown)"/>
      <param name="trust_distance" value="$(var trust_distance)"/>
      <param from="$(var sync_param_path)"/>
      <remap from="input" to="$(var input/clusters)"/>
      <remap from="output" to="$(var output/clusters)"/>

      <!-- rois, camera and info -->
      <param name="input/rois0" value="$(var input/rois0)"/>
      <param name="input/camera_info0" value="$(var input/camera_info0)"/>
      <param name="input/image0" value="$(var input/image0)"/>
      <param name="input/rois1" value="$(var input/rois1)"/>
      <param name="input/camera_info1" value="$(var input/camera_info1)"/>
      <param name="input/image1" value="$(var input/image1)"/>
      <param name="input/rois2" value="$(var input/rois2)"/>
      <param name="input/camera_info2" value="$(var input/camera_info2)"/>
      <param name="input/image2" value="$(var input/image2)"/>
      <param name="input/rois3" value="$(var input/rois3)"/>
      <param name="input/camera_info3" value="$(var input/camera_info3)"/>
      <param name="input/image3" value="$(var input/image3)"/>
      <param name="input/rois4" value="$(var input/rois4)"/>
      <param name="input/camera_info4" value="$(var input/camera_info4)"/>
      <param name="input/image4" value="$(var input/image4)"/>
      <param name="input/rois5" value="$(var input/rois5)"/>
      <param name="input/camera_info5" value="$(var input/camera_info5)"/>
      <param name="input/image5" value="$(var input/image5)"/>
      <param name="input/rois6" value="$(var input/rois6)"/>
      <param name="input/camera_info6" value="$(var input/camera_info6)"/>
      <param name="input/image6" value="$(var input/image6)"/>
      <param name="input/rois7" value="$(var input/rois7)"/>
      <param name="input/camera_info7" value="$(var input/camera_info7)"/>
      <param name="input/image7" value="$(var input/image7)"/>

      <!-- debug -->
      <param name="debug_mode" value="$(var debug_mode)"/>
      <param name="filter_scope_minx" value="$(var filter_scope_minx)"/>
      <param name="filter_scope_maxx" value="$(var filter_scope_maxx)"/>
      <param name="filter_scope_miny" value="$(var filter_scope_miny)"/>
      <param name="filter_scope_maxy" value="$(var filter_scope_maxy)"/>
      <param name="filter_scope_minz" value="$(var filter_scope_minz)"/>
      <param name="filter_scope_maxz" value="$(var filter_scope_maxz)"/>
      <param name="image_buffer_size" value="$(var image_buffer_size)"/>
    </node>
  </group>
</launch>
